### PR TITLE
[CMDUTILS][ATTRIB] Support folder attributes

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
JIRA issue: [CORE-16538](https://jira.reactos.org/browse/CORE-16538)

BEFORE:
attrib-before
AFTER:
after

### CODE REVIEW

**Comment by ThFabba** [comment]
While this fixes your specific example case, I think it will break a few others. Directories seem to be excluded whenever wildcards are used, so attributes are only applied to them if mentioned specifically by name.
Examples:
```
> mkdir folder1 folder2
> attrib +h folder*
--- yields: File not found - folder*
```

```
> mkdir folder1 folder2
> notepad file1.txt
> notepad folder1\file2.txt
> attrib +h /s
--- only applies the attribute to the two files, but not the folders
```

**Comment by katahiromz** [comment]
Ready to go.

File: base/applications/cmdutils/attrib/attrib.c [hunk 1]
```diff
@@ -180,75 +177,96 @@ ChangeAttribute(
...
-            ErrorMessage(GetLastError(), pszFile);
-            return 1;
+            dwAttribute = (dwAttribute & ~dwMask) | dwAttrib;
+            SetFileAttributes(szFullName, dwAttribute);
+            if (!bRecurse && !bDirectories)
+            {
+                ChangeAttribute(szFullName, L"*", dwMask, dwAttrib,
+                                bRecurse, FALSE);
+            }
+            if (bRecurse && !bDirectories)
```
Comment (ThFabba) [inline base/applications/cmdutils/attrib/attrib.c]: These could be `else if`s since the three conditions are mutually exclusive

Comment (katahiromz) [reply base/applications/cmdutils/attrib/attrib.c]: OK.

File: base/applications/cmdutils/attrib/attrib.c [hunk 2]
```diff
@@ -72,7 +72,7 @@ ErrorMessage(
                        NULL, dwErrorCode, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
                        (LPWSTR)&szError, 0, NULL))
     {
-        ConPrintf(StdErr, L"%s %s\n", szError, szMessage);
+        ConPrintf(StdOut, L"%s %s\n", szError, szMessage);
```
Comment (HBelusca) [inline base/applications/cmdutils/attrib/attrib.c]: Why?

File: base/applications/cmdutils/attrib/attrib.c [hunk 3]
```diff
@@ -285,7 +311,7 @@ int wmain(int argc, WCHAR *argv[])
         {
             if (wcslen(argv[i]) != 2)
             {
-                ConResPrintf(StdErr, STRING_ERROR_INVALID_PARAM_FORMAT, argv[i]);
+                ConResPrintf(StdOut, STRING_ERROR_INVALID_PARAM_FORMAT, argv[i]);
```
Comment (HBelusca) [inline base/applications/cmdutils/attrib/attrib.c]: Why?

File: base/applications/cmdutils/attrib/attrib.c [hunk 4]
```diff
@@ -312,15 +338,15 @@ int wmain(int argc, WCHAR *argv[])
                     break;
 
                 default:
-                    ConResPrintf(StdErr, STRING_ERROR_INVALID_PARAM_FORMAT, argv[i]);
+                    ConResPrintf(StdOut, STRING_ERROR_INVALID_PARAM_FORMAT, argv[i]);
```
Comment (HBelusca) [inline base/applications/cmdutils/attrib/attrib.c]: Why?

File: base/applications/cmdutils/attrib/attrib.c [hunk 5]
```diff
@@ -312,15 +338,15 @@ int wmain(int argc, WCHAR *argv[])
...
                     return -1;
             }
         }
         else if (*argv[i] == L'-')
         {
             if (wcslen(argv[i]) != 2)
             {
-                ConResPrintf(StdErr, STRING_ERROR_INVALID_PARAM_FORMAT, argv[i]);
+                ConResPrintf(StdOut, STRING_ERROR_INVALID_PARAM_FORMAT, argv[i]);
```
Comment (HBelusca) [inline base/applications/cmdutils/attrib/attrib.c]: Why?

File: base/applications/cmdutils/attrib/attrib.c [hunk 6]
```diff
@@ -371,20 +397,22 @@ int wmain(int argc, WCHAR *argv[])
...
-                         dwAttrib, bRecurse, bDirectories);
+        if (dwMask == 0)
+        {
+            PrintAttribute(szPath, szFileName, bRecurse);
+        }
+        else if (!ChangeAttribute(szPath, szFileName, dwMask,
+                                  dwAttrib, bRecurse, bDirectories))
+        {
+            LoadStringW(NULL, STRING_FILE_NOT_FOUND, szText, ARRAYSIZE(szText));
+            ConPrintf(StdOut, szText, argv[i]);
```
Comment (HBelusca) [inline base/applications/cmdutils/attrib/attrib.c]: `ConResPrintf(StdErr, STRING_FILE_NOT_FOUND, argv[i]);`

File: base/applications/cmdutils/attrib/attrib.c [hunk 7]
```diff
@@ -347,7 +373,7 @@ int wmain(int argc, WCHAR *argv[])
                     break;
 
                 default:
-                    ConResPrintf(StdErr, STRING_ERROR_INVALID_PARAM_FORMAT, argv[i]);
+                    ConResPrintf(StdOut, STRING_ERROR_INVALID_PARAM_FORMAT, argv[i]);
```
Comment (HBelusca) [inline base/applications/cmdutils/attrib/attrib.c]: Why?

File: base/applications/cmdutils/attrib/attrib.c (hunk 6 above)
Comment (katahiromz) [reply base/applications/cmdutils/attrib/attrib.c]: Fixed to `ConResPrintf(StdOut, STRING_FILE_NOT_FOUND, argv[i]);`.

File: base/applications/cmdutils/attrib/attrib.c (hunk 2 above)
Comment (katahiromz) [reply base/applications/cmdutils/attrib/attrib.c]: See `modules/rostests/apitests/cmd/cmd.c`, please.
The `attrib` program doesn't generate standard error messages.

**Pull request merged**